lib/dom.js

```diff
@@ -1175,12 +1189,23 @@ function serializeToString(node,buf,isHTML,nodeFilter,visibleNamespaces){
 		var prefixedNodeName = nodeName
 		if (!isHTML && !node.prefix && node.namespaceURI) {
 			var defaultNS
+			// lookup current default ns from `xmlns` attribute
```

Comment:
minor nit: I wonder if we could somehow use Array.prototype.find() here, assuming we can start dropping some extra-old browsers?

Reply:
Currently we support ES5, requiring Array.find to be available would be a massive drop of supported (old) versions as far as I can judge. And I'm not aware how could measure/judge when to do that.
I think we should discuss how and when to drop support (e.g. in the related runtimes discussion). But not "now".